build/conditional-container-push.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+
+usage() {
+    echo "Usage: $0 IMAGE_URI" >&2
+    exit 1
+}
+
+## image_exists_in_repo IMAGE_URI
+#
+# Checks whether IMAGE_URI -- e.g. quay.io/app-sre/osd-metrics-exporter:abcd123
+# -- exists in the remote repository.
+# If so, returns success.
+# If the image does not exist, but the query was otherwise successful, returns
+# failure.
+# If the query fails for any reason, prints an error and *exits* nonzero.
+#
+# This function cribbed from:
+# https://github.com/openshift/boilerplate/blob/d2d6640ef28a7ce54ac639db32c825aa4bb492a0/boilerplate/_lib/common.sh#L77-L125
+image_exists_in_repo() {
+    local image_uri=$1
+    local output
+
+    if output=$(skopeo inspect docker://"${image_uri}" 2>&1); then
```

Comment:
We need some way to install the `skopeo` binary, no?

Reply:
Oh, that's a fair point :P

Do you know of a way I can play with this `ubuntu-1604:201903-01` machine image somehow? [Installing skopeo](https://github.com/containers/skopeo/blob/master/install.md) is, for some reason, not a simple thing on ubuntu, so I would want a way to test out whatever I come up with.

(By the way, I would use `docker image inspect` instead, except that requires pulling the image locally, whereas skopeo works remote. Other options include making direct REST queries to the docker API on quay, e.g. via python.)

Reply:
I always use `docker image inspect` as well. I don't think pulling the image down locally is a deal breaker. If it's simpler, let's go with that :+1:

Reply:
Okay, did that. Put it in a separate commit, mostly so future-me can see that delta.